lib/pbench/server/auth/__init__.py

```diff
@@ -33,18 +35,18 @@ def __init__(
         server_url: str,
         realm_name: str,
         client_id: str,
-        logger,
+        logger: logging.Logger,
         client_secret_key: str = None,
         verify: bool = True,
-        headers: Dict = None,
+        headers: Optional[Dict[str, str]] = None,
         timeout: int = 60,
     ):
         self.server_url = server_url
         self.client_id = client_id
         self.client_secret_key = client_secret_key
         self.realm_name = realm_name
         self.logger = logger
-        self.headers = headers if headers is not None else dict()
+        self.headers = headers if headers is not None else CaseInsensitiveDict()
```

Comment:
As Webb pointed out in my client API, this leaves `self.headers` as a `Dict` if one was supplied by the caller and `CaseInsensitiveDict()` otherwise. While the latter is a subclass of the former, the case-insensitivity is fairly important for headers, and it probably makes more sense to use a pattern such as Webb suggested in that context, e.g.,

```
    self.headers = CaseInsensitiveDict()
    if headers:
        self.headers.update(headers)
```